# Preliminary Individual DEA

## Purpose

This notebook is to explore the given data set individually to set the stage for group work discussion for the end of module 1 project of Flatiron School Data Science Immersive 091420-Manhattan Cohorrt. The notebook will serve as both code reference in the future and surface level data exploration of the given data set.

## Goal

The provided data set are series of information regarding movies. They will be explored to provide insight into current trend and success parameters to recommend a future course of movie creation of Microsoft Movie Studios. 

## Structure

The data set can be found in `./data` folder 
<br>
The data is pulled from following companies: 

- Box Office Mojo (prefix: bom)
- IMDB (prefix: imdb)
- Rotten Tomatoes (prefix: rt)
- TheMoiveDB.org (prefix: tmdb) (very confusing with imdb...)


### Contribution
Author - Yung Han Jeong
Team Member - Cierra Andaur

#### Library Import

In [12]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [5]:
!ls data

bom.movie_gross.csv.gz       imdb.title.ratings.csv.gz
imdb.name.basics.csv.gz      rt.movie_info.tsv.gz
imdb.title.akas.csv.gz       rt.reviews.tsv.gz
imdb.title.basics.csv.gz     tmdb.movies.csv.gz
imdb.title.crew.csv.gz       tn.movie_budgets.csv.gz
imdb.title.principals.csv.gz


#### Box Office Mojo

In [33]:
bomgr_df = pd.read_csv("data/bom.movie_gross.csv.gz")
bomgr_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [34]:
bomgr_df.corr()

domestic_gross      year
domestic_gross        1.000000  0.018708
year                  0.018708  1.000000

#### IMDB

IMDB data set is very separated, but provides with given key.

- imdb.name.basics: name of actor/staff memberds. ID'ed by nconst column and title_id list in known_for titles column
- imdb.title.akas: alternaitve tiltes of movies (title_id_ depending on the region)
- imdb.title.basics: movie title with basic info
- imdb.title.crew: list of directors and writers but ID'ed with nconst
- imdb.principals: principals actors/staffs? "Featuring"?
- imdb.rating: rating and number of reviews

In [46]:
i_basic_df = pd.read_csv("data/imdb.name.basics.csv.gz")
i_basic_df.head()

nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

In [39]:
i_aka_df = pd.read_csv("data/imdb.title.akas.csv.gz")
i_aka_df.head()

title_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   

  language        types   attributes  is_original_title  
0       bg          NaN          NaN                0.0  
1      NaN  imdbDisplay          NaN                0.0  
2      NaN  imdbDisplay          NaN                0.0  
3      NaN          NaN  short title                0.0  
4      NaN  imdbDisplay          NaN                0.0

In [50]:
i_title_basic_df = pd.read_csv("data/imdb.title.basics.csv.gz")
i_title_basic_df.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [45]:
i_crew_df = pd.read_csv("data/imdb.title.crew.csv.gz")
i_crew_df.head()

tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

In [41]:
i_principal_df = pd.read_csv("data/imdb.title.principals.csv.gz")
i_principal_df.head()

tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

In [37]:
i_rating_df = pd.read_csv("data/imdb.title.ratings.csv.gz")
i_rating_df.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

### Rotten Tomatoes

- rt.movie_info: informatino regarding the movies
- rt.reviews: reviews, but most ratings are in fresh or rotten

In [54]:
rt_info_df = pd.read_csv("data/rt.movie_info.tsv.gz", delimiter = "\t", encoding='ISO-8859-1' )
#thank you Tim Hintz and Eric Roberts for encoding info 
rt_info_df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [55]:
rt_review_df = pd.read_csv("data/rt.reviews.tsv.gz", delimiter = "\t", encoding='ISO-8859-1' )
rt_review_df.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [88]:
rt_review_df.fresh.value_counts()

fresh     33035
rotten    21397
Name: fresh, dtype: int64

Adding fresh_bool column to assign 1 or fresh 0 or rotten

In [91]:
rt_review_df["fresh_bool"] = rt_review_df["fresh"].apply(lambda txt: 1 if txt == "fresh" else 0)

In [93]:
rt_review_df.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  fresh_bool  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018           1  
1  Annalee Newitz           0           io9.com       May 23, 2018           0  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018           1  
3   Daniel Kasman           0              MUBI  November 16, 2017           1  
4             NaN           0      Cinema Scope   October 12, 2017           1

### The Movie Database

In [57]:
tmdb_df = pd.read_csv("data/tmdb.movies.csv.gz")
tmdb_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

### TN is most likely to be "The Numbers" the-numbers.com

In [66]:
tn_budget_df = pd.read_csv("data/tn.movie_budgets.csv.gz")
tn_budget_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [68]:
tn_budget_df.info() #all budgets are strings initially

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
id                   5782 non-null int64
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null object
domestic_gross       5782 non-null object
worldwide_gross      5782 non-null object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [72]:
#using code below for number conversion
int("".join(x for x in "$410,600,000" if x.isnumeric()))

410600000

Below changes each columsn to numberic(int64) values.
<br>
#### REMINDER: BELOW CAN ONLY BE RAN ONCE SINCE IT REPLACES THE COLUMN. 
#### IF STRING VALUES ARE NEEDED REASSIGN COLUMNS OR RELOAD DATA

In [80]:
tn_budget_df["production_budget"] = tn_budget_df["production_budget"].apply(lambda num: int("".join(x for x in num if x.isnumeric())))

In [78]:
tn_budget_df["domestic_gross"] = tn_budget_df["domestic_gross"].apply(lambda num: int("".join(x for x in num if x.isnumeric())))

In [79]:
tn_budget_df["worldwide_gross"] = tn_budget_df["worldwide_gross"].apply(lambda num: int("".join(x for x in num if x.isnumeric())))

In [81]:
tn_budget_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
id                   5782 non-null int64
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null int64
domestic_gross       5782 non-null int64
worldwide_gross      5782 non-null int64
dtypes: int64(4), object(2)
memory usage: 271.2+ KB


In [82]:
tn_budget_df.corr() #much better

id  production_budget  domestic_gross  \
id                 1.000000          -0.035278        0.008255   
production_budget -0.035278           1.000000        0.685682   
domestic_gross     0.008255           0.685682        1.000000   
worldwide_gross   -0.009422           0.748306        0.938853   

                   worldwide_gross  
id                       -0.009422  
production_budget         0.748306  
domestic_gross            0.938853  
worldwide_gross           1.000000

According to the correlation table above:

- The movie budget gas a positive correlation to the gross numbers.
- If a movie does well in domestic market, it is very likely to do well in worldwide market (0.94!)